packages/yarnpkg-core/sources/formatUtils.ts

```diff
@@ -249,6 +265,31 @@ const transforms = {
     },
   }),
 
+  [Type.SIZE_DIFF]: validateTransform({
+    pretty: (configuration: Configuration, size: number) => {
+      const absSize = Math.abs(size);
+      const thresholds = [`KiB`, `MiB`, `GiB`, `TiB`];
+
+      let power = thresholds.length;
+      while (power > 1 && absSize < 1024 ** power)
+        power -= 1;
+
+      const factor = 1024 ** power;
+      const value = Math.floor(absSize * 100 / factor) / 100;
+
+      const sign = size >= 0 ? `+` : `-`;
```

Comment:
I think `0` are best handled by not showing the diff at all when there are none - that's the approach taken in the cache message. I'll however tweak it to always show at least `+ 0.01 KiB` or `- 0.01 KiB`, since a rounding error could lead to a surprising `+ 0.00 KiB`.